tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -86,652 +85,952 @@
+                    scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
+            );
+            List<TopicPartitionInfo>  partitions = adminClient
+                    .describeTopics(Collections.singletonList(testTopicName))
+                    .allTopicNames()
+                    .get()
+                    .get(testTopicName)
+                    .partitions();
+            assertEquals(2, partitions.size(), "Unequal partition size: " + partitions.size());
+            assertEquals(defaultReplicationFactor, (short) partitions.get(0).replicas().size(), "Unequal replication factor: " + partitions.get(0).replicas().size());
+        }
+    }
 
-        TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, 2, 2,
-                scala.collection.immutable.Map$.MODULE$.empty(), topicConfig
-        );
+    @ClusterTemplate("generate1")
+    public void testCreateWithDefaultPartitions() throws InterruptedException, ExecutionException {
+        String testTopicName = org.apache.kafka.test.TestUtils.getCurrentFunctionName() + "-" +
+                org.apache.kafka.test.TestUtils.randomString(10);
+
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            Buffer<KafkaBroker> scalaBrokers = JavaConverters.asScalaIteratorConverter(
+                            clusterInstance.brokers().values().iterator())
+                    .asScala().toBuffer();
+            Seq<ControllerServer> scalaControllers = JavaConverters.asScalaIteratorConverter(
+                            clusterInstance.controllers().values().iterator())
+                    .asScala().toSeq();
+            TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, defaultNumPartitions, 2,
+                    scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
+            );
+            List<TopicPartitionInfo> partitions = adminClient
+                    .describeTopics(Collections.singletonList(testTopicName))
+                    .allTopicNames()
+                    .get()
+                    .get(testTopicName)
+                    .partitions();
+
+            assertEquals(defaultNumPartitions, partitions.size(), "Unequal partition size: " + partitions.size());
+            assertEquals(2, (short) partitions.get(0).replicas().size(), "Partitions not replicated: " + partitions.get(0).replicas().size());
+        }
+    }
 
-        Config configs = adminClient.describeConfigs(Collections.singleton(configResource)).all().get().get(configResource);
-        assertEquals(1000, Integer.valueOf(configs.get("delete.retention.ms").value()),
-                "Config not set correctly: " + configs.get("delete.retention.ms").value());
+    @ClusterTemplate("generate1")
+    public void testCreateWithConfigs() throws Exception {
+        String testTopicName = org.apache.kafka.test.TestUtils.getCurrentFunctionName() + "-" +
```

Comment:
It seems to me `randomString` is good enough. Or we can add a static count to generate random topic name

Reply:
My consideration is the FunctionName can help us to address which testcase fail.

Reply:
or we can inject `TestInfo` to get name?